collect_app/src/main/java/org/odk/collect/android/preferences/SubmissionsSharedPreferences.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2018 Nafundi
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.odk.collect.android.preferences;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+
+import org.odk.collect.android.application.Collect;
+
+import java.util.Map;
+
+public class SubmissionsSharedPreferences {
+
+    public static final String SUBMISSIONS_PREFERENCES = "submissions_preferences";
+    public static final String KEY_SUBMISSION = "submissions_list_key_";
+
+    private final SharedPreferences sharedPreferences;
+
+    private static SubmissionsSharedPreferences instance;
+
+    private SubmissionsSharedPreferences() {
+        sharedPreferences = Collect.getInstance().getSharedPreferences(SUBMISSIONS_PREFERENCES, Context.MODE_PRIVATE);
+    }
+
+    public static synchronized SubmissionsSharedPreferences getInstance() {
```

Comment:
I used Dagger to create the submission manager so that testing would be alot easier in the sense that we wouldn't manually be creating singletons. There's a `SharedPreferences` cleanup being done by @shobhitagarwal1612 that's gonna help with this as well. Nonetheless, let me know what you think.

Reply:
@grzesiek2010 do you want any help with this one? or you gonna just make the changes? Let me know.

Reply:
Sorry I didn't have time to review your comments since I was rather focused on reviewing other prs (not mine) during @lognaturel's brake. As I said above I made that change to make the code consistent. I have to investigate those tests but I'm not sure if I'm able to do that today and I have a break till Monday so maybe on Monday.

Reply:
Okay no problem! Have a good break!